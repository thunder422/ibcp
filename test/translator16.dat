###################################################
#  Translator Test #15: Colon Tests (2013-04-05)  #
###################################################

A=SQR(17):PRINT A
PRINT A:PRINT B
PRINT A;:PRINT B
INPUT A:PRINT A
A,B,C=5.0:INPUT D
A$,LEFT$(B$,1),MID$(C$,2,3)="A":PRINT A$
INPUT PROMPT "Number",A:PRINT A
INPUT PROMPT "Number";A:PRINT A
INPUT PROMPT "Number:";A:PRINT A
INPUT A;:PRINT A
PRINT A:REM output A
PRINT A:'output A:test
INPUT A;:R=SQR(A):PRINT"Sqrt(";A;")=";R

# colon error tests
:
PRINT:
PRINT::PRINT
