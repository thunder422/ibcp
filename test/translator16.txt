
Input: A=SQR(17):PRINT A
Output: A<ref> 17 SQR( Assign':' A PrintDbl PRINT 

Input: PRINT A:PRINT B
Output: A PrintDbl PRINT':' B PrintDbl PRINT 

Input: PRINT A;:PRINT B
Output: A PrintDbl ;':' B PrintDbl PRINT 

Input: INPUT A:PRINT A
Output: InputParse InputBegin A<ref> InputAssign INPUT':' A PrintDbl PRINT 

Input: A,B,C=5.0:INPUT D
Output: A<ref> B<ref> C<ref> 5.0 AssignList':' InputParse InputBegin D<ref> InputAssign INPUT 

Input: A$,LEFT$(B$,1),MID$(C$,2,3)="A":PRINT A$
Output: A$<ref> B$<ref> 1% C$<ref> 2% 3% "A" AssignKeepMid3 AssignKeepLeft Assign$':' A$ PrintStr PRINT 

Input: INPUT PROMPT "Number",A:PRINT A
Output: "Number" InputParse InputBeginStr'Question' A<ref> InputAssign INPUT-PROMPT':' A PrintDbl PRINT 

Input: INPUT PROMPT "Number";A:PRINT A
Output: "Number" InputParse InputBeginStr A<ref> InputAssign INPUT-PROMPT':' A PrintDbl PRINT 

Input: INPUT PROMPT "Number:";A:PRINT A
Output: "Number:" InputParse InputBeginStr A<ref> InputAssign INPUT-PROMPT':' A PrintDbl PRINT 

Input: INPUT A;:PRINT A
Output: InputParse InputBegin A<ref> InputAssign INPUT'Keep:' A PrintDbl PRINT 

Input: PRINT A:REM output A
Output: A PrintDbl PRINT':' REM| output A| 

Input: PRINT A:'output A:test
Output: A PrintDbl PRINT':' '|output A:test| 

Input: INPUT A;:R=SQR(A):PRINT"Sqrt(";A;")=";R
Output: InputParse InputBegin A<ref> InputAssign INPUT'Keep:' R<ref> A SQR( Assign':' "Sqrt(" PrintStr A PrintDbl ")=" PrintStr R PrintDbl PRINT 

Input: :
       ^-- expected command

Input: PRINT:
             ^-- expected command

Input: PRINT::PRINT
             ^-- expected command

