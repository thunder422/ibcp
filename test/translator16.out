
Input: A=SQR(17):PRINT A
Output: A = SQR(17): PRINT A 

Input: PRINT A:PRINT B
Output: PRINT A: PRINT B 

Input: PRINT A;:PRINT B
Output: PRINT A;: PRINT B 

Input: INPUT A:PRINT A
Output: INPUT A: PRINT A 

Input: A,B,C=5.0:INPUT D
Output: A, B, C = 5.0: INPUT D 

Input: A$,LEFT$(B$,1),MID$(C$,2,3)="A":PRINT A$
Output: A$, LEFT$(B$, 1), MID$(C$, 2, 3) = "A": PRINT A$ 

Input: INPUT PROMPT "Number",A:PRINT A
Output: INPUT PROMPT "Number", A: PRINT A 

Input: INPUT PROMPT "Number";A:PRINT A
Output: INPUT PROMPT "Number"; A: PRINT A 

Input: INPUT PROMPT "Number:";A:PRINT A
Output: INPUT PROMPT "Number:"; A: PRINT A 

Input: INPUT A;:PRINT A
Output: INPUT A;: PRINT A 

Input: PRINT A:REM output A
Output: PRINT A: REM output A 

Input: PRINT A:'output A:test
Output: PRINT A: 'output A:test 

Input: INPUT A;:R=SQR(A):PRINT"Sqrt(";A;")=";R
Output: INPUT A;: R = SQR(A): PRINT "Sqrt("; A; ")="; R 

Input: :
       ^-- expected command or item for assignment

Input: PRINT:
             ^-- expected command or item for assignment

Input: PRINT::PRINT
             ^-- expected command or item for assignment

